api-report/core-interfaces.api.md

```diff
@@ -4,6 +4,13 @@
 
 ```ts
 
+// @public
+export function extractLogSafeErrorProperties(error: any, sanitizeStack: boolean): {
```

Comment:
we shouldn't put functions, or any code really in our common/definitions packages, they should jsut be types and interfaces

Reply:
I believe this is used transitively by one of the migrated classes. I'll double check the exact usages.

Reply:
i think the error implementations should probably move to telemetry-utils, since we don't use implementations in our definitions, so it should be fine

Reply:
The types we were discussing moving from `container-utils` are classes that extend a shared base class (currently in telemetry-utils). That base class uses the moved functions directly or transitively. If we don't want to move functions and classes here, then we'll need to figure out a new home for those error types. Any recommendations?

For reference, the dependency graph of the packages we've been discussing looks like:
```mermaid
flowchart BT
container-definitions --> core-interfaces
telemetry-utils --> core-interfaces
container-utils --> container-definitions
container-utils --> telemetry-utils
```

Reply:
i think we should move implementation to telemetry-utils, and types used in our exported surface area to core-interfaces. Telemetry is an area where we have simple external/exported types, but more complex internal types and implementations.

Reply:
i haven't tried to move everything, so maybe those goals fall apart somewhere, easiest way to find out is usually to just try and move stuff